engine/namegen.ml

```diff
@@ -308,6 +308,13 @@ let visible_ids sigma (nenv, c) =
       | _ -> ids
       in
       accu := (gseen, vseen, ids)
+  | exception Not_found when !Flags.in_debugger ->
+    (* Happens in debugger where the correct sigma is unknown *)
+    SList.Skip.iter (fun c -> visible_ids n (EConstr.of_constr c))
+      (snd (Constr.destEvar (EConstr.Unsafe.to_constr c)))
+  | Evar (_,args) ->
+    (* Useful for debugger: bypass iter_with_binders which requires a correct sigma *)
+    SList.Skip.iter (visible_ids n) args
```

Comment:
Since this is done unconditionally I think it could lead to alpha renaming bugs in normal operation. Maybe we should instead do
```suggestion
  | Evar ev ->
    begin match Evd.expand_existential sigma ev with
    | args -> List.iter (visible_ids n) args
    | exception Not_found -> (* Useful for debugger: bypass iter_with_binders which requires a correct sigma *)
      SList.Skip.iter (visible_ids n) args
    end
```

Reply:
I see. I overlooked that `expand_existential` was about expanding the arguments.